src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.mux.cs

```diff
@@ -0,0 +1,2673 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// MUX Reference controls\dev\TeachingTip\TeachingTip.cpp, tag winui3/release/1.5.0, commit c8bd154c0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Microsoft.UI.Private.Controls;
+using Uno.Disposables;
+using Uno.Extensions;
+using Uno.UI.DataBinding;
+using Uno.UI.Helpers.WinUI;
+using Windows.ApplicationModel;
+using Windows.Foundation;
+using Windows.Foundation.Metadata;
+using Windows.Graphics.Display;
+using Windows.System;
+using Windows.UI.Core;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Automation;
+using Microsoft.UI.Xaml.Automation.Peers;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Shapes;
+using static Uno.UI.Helpers.WinUI.CppWinRTHelpers;
+using CompositionTarget = Microsoft.UI.Xaml.Media.CompositionTarget;
+using Microsoft.UI.Composition;
+
+namespace Microsoft.UI.Xaml.Controls;
+
+public partial class TeachingTip : ContentControl
+{
+	private const string c_TitleTextBlockVisibleStateName = "ShowTitleTextBlock";
+	private const string c_TitleTextBlockCollapsedStateName = "CollapseTitleTextBlock";
+	private const string c_SubtitleTextBlockVisibleStateName = "ShowSubtitleTextBlock";
+	private const string c_SubtitleTextBlockCollapsedStateName = "CollapseSubtitleTextBlock";
+
+	private const string c_OverlayCornerRadiusName = "OverlayCornerRadius";
+
+	/// <summary>
+	/// Initializes a new instance of the TeachingTip class.
+	/// </summary>
+	public TeachingTip()
+	{
+		DefaultStyleKey = typeof(TeachingTip);
+		// EnsureProperties()
+		Unloaded += ClosePopupOnUnloadEvent;
+		m_automationNameChangedRevoker = RegisterPropertyChangedCallback(AutomationProperties.NameProperty, OnAutomationNameChanged);
+		m_automationIdChangedRevoker = RegisterPropertyChangedCallback(AutomationProperties.AutomationIdProperty, OnAutomationIdChanged);
+		SetValue(TemplateSettingsProperty, new TeachingTipTemplateSettings());
+	}
+
+	protected override AutomationPeer OnCreateAutomationPeer() =>
+		new TeachingTipAutomationPeer(this);
+
+	protected override void OnApplyTemplate()
+	{
+		base.OnApplyTemplate();
+
+		m_acceleratorKeyActivatedRevoker.Disposable = null;
+		m_previewKeyDownForF6Revoker.Disposable = null;
+		m_effectiveViewportChangedRevoker.Disposable = null;
+		m_contentSizeChangedRevoker.Disposable = null;
+		m_closeButtonClickedRevoker.Disposable = null;
+		m_alternateCloseButtonClickedRevoker.Disposable = null;
+		m_actionButtonClickedRevoker.Disposable = null;
+
+		m_container = (Border)GetTemplateChild(s_containerName);
+		m_rootElement = m_container.Child;
+		m_rootElement.FreezeTemplatedParent();
```

Comment:
Let's add an "Uno specific" comment with link issue so we can remove this workaround when the issue is fixed in future.